lib/bap_llvm/bap_llvm_ogre_coff.ml

```diff
@@ -69,9 +96,37 @@ module Relocatable = struct
         ~f:(fun (name,_,_,off) (_,_,size) -> name,off,size) >>= fun s ->
       Fact.Seq.iter s
         ~f:(fun (name, off, size) ->
-            let addr = Int64.(base + off) in
-            Fact.provide section addr size >>= fun () ->
-            Fact.provide named_region addr size name)
+          let addr = Int64.(base + off) in
+          Fact.provide section addr size >>= fun () ->
+          Fact.provide named_region addr size name)
+
+    let code =
+      Fact.require base_address >>= fun base ->
+      Fact.foreach Ogre.Query.(
+        select (from section_entry $ code_entry)
+          ~join:[[field name];
+                 [field size ~from:section_entry;
+                  field size ~from:code_entry]])
+        ~f:(fun _ (_,off,size) -> Int64.(base + off),size,off) >>= fun s ->
+      Fact.Seq.iter s
+        ~f:(fun (addr, size, off) ->
+          Fact.provide code_region addr size off)
+
+    let data =
+      Fact.require base_address >>= fun base ->
+      Fact.foreach Ogre.Query.(
+        select (from section_entry $ data_entry)
+          ~join:[[field name];
+                 [field size ~from:section_entry;
+                  field size ~from:data_entry]])
+        ~f:(fun _ (_,off,size,w) -> Int64.(base + off),size,off,w) >>= fun s ->
+      Fact.Seq.iter s
+        ~f:(fun (addr, size, off, w) ->
+          Fact.provide data_region addr size off w)
```

Comment:
so those two (nearly identical, again) functions just map `code_entry -> code_region` and `data_entry -> data_region`, right?